src/modules/aws_ec2/entity/key_pair.ts

```diff
@@ -0,0 +1,44 @@
+import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn, Unique } from 'typeorm';
+
+import { KeyType } from '@aws-sdk/client-ec2';
+
+import { cloudId } from '../../../services/cloud-id';
+import { AwsRegions } from '../../aws_account/entity';
+
+@Entity()
+@Unique('uq_keypair_id_region', ['id', 'region'])
+@Unique('uq_keypair_name_region', ['name', 'region'])
+export class KeyPair {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  @cloudId
+  name: string;
+
+  @Column({ nullable: true })
+  keyPairId?: string;
```

Comment:
Maybe this should be `@cloudId` instead?

Reply:
is the `name` unique for AWS? Usually what I do is set the cloudId as the value we use for the SDK GET method

Reply:
it is unique, i cannot create 2 keys with the same name. It is also the method i use for GET